### PR TITLE
chore: fix incorrect locale in vite

### LINKED ISSUE #39045: 打包后 语言包丢失 开发的为中文 打包后变英文了
### Reproduction link

[https://github.com/muddyrainTestProject/antdv5.0.2-test](https://github.com/muddyrainTestProject/antdv5.0.2-test)

### Steps to reproduce

引入 语言包 
上下文引入 然后 嵌套 APP

### What is expected?

打包后会根据引入的语言包来进行打包

### What is actually happening?

引入的中文语言包 打包后为中文

| Environment | Info |
| --- | --- |
| antd | 5.0.2 |
| React | react |
| System |  macOS 12.3  |
| Browser | chrome 106.0.5249.119 |

### PR DESCRIPTION
**Pull request by Dunqing**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix: #39045 
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix use antd/locale/*.js not work      |
| 🇨🇳 Chinese |    修复使用antd/locale/*.js不生效的       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by wkl007** [comment]
大佬们，这个 PR 还没有合并吗，这个 issue https://github.com/ant-design/ant-design/issues/38692 已经好久了。

**Comment by yoyo837** [comment]
> 大佬们，这个 PR 还没有合并吗，这个 issue #38692 已经好久了。

这个方案可能有待斟酌

**Comment by MadCcc** [comment]
暂时先不考虑 exports 方案了，手动维护比较麻烦，vite 用户引导到 FAQ

**Comment by yoyo837** [comment]
体验起来...

File: package.json [hunk 1]
```diff
@@ -44,6 +44,20 @@
...
+  "exports": {
+    ".": {
+      "import": "./es/index.js",
+      "require": "./lib/index.js",
+      "types": "./es/index.d.ts"
+    },
+    "./locale/*": {
+      "import": "./es/locale/*.js",
+      "require": "./lib/locale/*.js",
+      "default": "./locale/*.js",
```
Comment (PeachScript) [inline package.json:56]: 这里是不是不需要了，有 exports 以后产物根目录应该不需要包含 locale 文件夹了？

Comment (Dunqing) [reply package.json:56]: 为了避免 breaking changes 应该会保留原有的所有文件

**Comment by li-jia-nan** [comment]
conflicts

**Pull request closed without merging**